cli/sync_service/object.go

```diff
@@ -5,104 +5,170 @@ import (
 	"github.com/open-horizon/anax/config"
-	"github.com/open-horizon/anax/exchange"
 	"github.com/open-horizon/anax/i18n"
 	"github.com/open-horizon/edge-sync-service/common"
 	"net/http"
 	"os"
 	"path"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
 )
 
 type MMSObjectInfo struct {
-	Definition   common.MetaData             `json:"definition"`
+	ObjectID     string                      `json:"objectID,omitempty"`
+	ObjectType   string                      `json:"objectType,omitempty"`
+	Definition   *common.MetaData            `json:"definition,omitempty"`
 	Destinations []common.DestinationsStatus `json:"destinations,omitempty"`
+	ObjectStatus string                      `json:"objectStatus,omitempty"`
 }
 
-// Display the object metadata for a given object in the MMS.
-func ObjectList(org string, userPw string, objType string, objId string, details bool) {
+// Display the object metadata for given flags in the MMS.
+func ObjectList(org string, userPw string, objType string, objId string, destPolicy string, dpService string, dpPropertyName string, dpUpdateTimeSince string, destType string, destId string, withData string, expirationTimeBefore string, long bool, details bool) {
 	// get message printer
 	msgPrinter := i18n.GetMessagePrinter()
 
 	if userPw == "" {
 		cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("must specify exchange credentials to access the model management service."))
 	}
 
-	if details && objId == "" {
-		cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("must specify object ID when requesting object details."))
-	}
-
 	// Set the API key env var if that's what we're using.
 	cliutils.SetWhetherUsingApiKey(userPw)
 
-	// If object ID is omitted, query all objects of the given type.
-	if !details {
-		objectList := new(exchange.ObjectDestinationPolicies)
-		urlPath := path.Join("api/v1/objects/", org, objType, "?all_objects=true")
+	var objectsMeta []common.MetaData
+
+	// validate params
+	if strings.ToLower(destPolicy) != "true" && strings.ToLower(destPolicy) != "false" {
+		destPolicy = ""
+	} else {
+		destPolicy = strings.ToLower(destPolicy)
+	}
+
+	noData := ""
+	if strings.ToLower(withData) == "true" || strings.ToLower(withData) == "false" {
+		withDataBool, _ := strconv.ParseBool(strings.ToLower(withData))
+		noData = strconv.FormatBool(!withDataBool)
+	}
 
-		// Call the MMS service over HTTP to get the basic object metadata.
-		httpCode := cliutils.ExchangeGet("Model Management Service", cliutils.GetMMSUrl(), urlPath, cliutils.OrgAndCreds(org, userPw), []int{200, 404}, objectList)
-		if httpCode == 404 {
-			cliutils.Fatal(cliutils.NOT_FOUND, msgPrinter.Sprintf("no objects type '%s' found in org %s", objType, org))
+	if dpService != "" || dpPropertyName != "" || dpUpdateTimeSince != "" {
+		if destPolicy == "false" {
+			cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("must specify destinationPolicy to true or set to empty when filter by --service, --property, or --updateTime"))
```

Comment:
Please reword:
must omit --policy or set it to true when filtering by --service, --property, or --updateTime"